File: library/src/androidTest/java/hotchemi/android/rate/UriHelperTest.java
package hotchemi.android.rate;

import android.net.Uri;
import android.test.AndroidTestCase;

import hotchemi.android.rate.internal.UriHelper;

/**
 * Unit test for {@link UriHelper}
 */
public class UriHelperTest extends AndroidTestCase {

    private static final String GOOGLE_PLAY = "https://play.google.com/store/apps/details?id=";

    public void testGetGooglePlayUri() {
        {
            Uri uri = UriHelper.getGooglePlay("");
            assertEquals(uri.toString(), GOOGLE_PLAY);
        }
        {
            Uri uri = UriHelper.getGooglePlay(null);
            assertNull(uri);
        }
        {
            final String packageName = "hotchemi.android.rate";
            Uri uri = UriHelper.getGooglePlay(packageName);
            assertEquals(uri.toString(), GOOGLE_PLAY + packageName);
        }
    }
}

File: library/src/test/java/hotchemi/android/rate/internal/UriHelperTest.kt
package hotchemi.android.rate.internal

import com.google.common.truth.Truth.assertThat
import org.junit.Test
import org.junit.runner.RunWith
import org.robolectric.RobolectricTestRunner

@RunWith(RobolectricTestRunner::class)
class UriHelperTest {

    @Test
    fun `getGooglePlay should append non-null package name to id param`() {
        val actual = UriHelper.getGooglePlay("com.example.app").toString()

        assertThat(actual).isEqualTo("https://play.google.com/store/apps/details?id=com.example.app")
    }

    @Test
    fun `getAmazonAppstore should append non-null package name to id param`() {
        val actual = UriHelper.getAmazonAppstore("com.example.app").toString()

        assertThat(actual).isEqualTo("amzn://apps/android?p=com.example.app")
    }
}